### DISCUSSION TITLE
Istio-proxy pods consuming high cpu for few hundreds of requests

### DISCUSSION THREAD

**Discussion by krishna-aw** [istio/istio#49967] [Q&A]
istio version: 1.15.7
EKS: 1.26

The istio-proxy pods are consuming lots of CPU for few hundreds(~400) of requests. We have a HPA setup so that if CPU utilization reaches 80%, they scale up. Due to this several replicas(~30) are created. How do I troubleshoot to find the source of high CPU usage?  

Here is the number of requests graph:
![Screenshot 2024-03-18 at 1 59 46 PM](https://github.com/istio/istio/assets/139287417/2cb07231-381b-4198-ac26-67da68fbc83d)
The CPU usage graph
![Screenshot 2024-03-18 at 1 59 04 PM](https://github.com/istio/istio/assets/139287417/ffc35f5a-b5e4-42df-8f42-d3407d376bd9)
Number of Pods scaled up by HPA during this time period
![Screenshot 2024-03-18 at 1 59 27 PM](https://github.com/istio/istio/assets/139287417/51e3de1c-18d7-44ee-9e40-450df8541e24)

**Comment by zirain** [comment]
can you check the number of stats by running `istioctl x es <podname>.<namespace> -oprom|wc -l`
